### ISSUE TITLE
Player config that uses vocal score crashes when vocal score audio doesn't exist.

### ISSUE THREAD

**Issue by kyzentun** [dguzek/Simply-Love-SM5#19]
I updated to a new version of SL and didn't copy the vocal score stuff over, but my player config is still set to use it.  So when I got to ScreenEvaluation, it crashed trying to find the theme element.

17:45.453: Exception: Theme element "Other/Vocalize/Shizuka (Japanese)/default.lua" could not be found in "Themes/Simply-Love-SM5/" or "Themes/_fallback/".

Pass true as the third argument to THEME:GetPathO to indicate that the element is optional.  You'll get an empty string if the thing isn't found.  This works for all variants of GetPath.

**Comment by kyzentun** [comment]
Too lazy to submit a PR right now, the important part is this:
```
				local voice_path= THEME:GetPathO("","Vocalize/" .. voice .. "/default.lua", true)
				if voice_path ~= "" then
```

You can use the patch command line tool to apply a diff, if patch exists on OS X.  ```-p1``` tells it the level of the patch, and ```vocal_fix``` is the file that is being directed to patch's input.  So stick the diff below in ```vocal_fix``` in the top level SL dir and the command should work.

patch -p1 < vocal_fix
```
diff --git a/BGAnimations/ScreenEvaluation common/score_vocalization.lua b/BGAnimations/ScreenEvaluation common/score_vocalization.lua
index 029ace4..dfb2540 100644
--- a/BGAnimations/ScreenEvaluation common/score_vocalization.lua	
+++ b/BGAnimations/ScreenEvaluation common/score_vocalization.lua	
@@ -50,31 +50,39 @@ return Def.Actor{
 			if SL[pn].ActiveModifiers.Vocalization == "Blender" then
 				voice = RandomizeVocalization()
 			end
-			
+
+			local voice_loaded= false
 			if not Vocalization[voice] then
 				-- load the voice timings
-				dofile(THEME:GetPathO("","Vocalize/" .. voice .. "/default.lua"))
-			end
-			
-			local number = Digits[ActiveDigit][1]
-			local soundbyte = GetVocalizeDir() .. voice .. "/" .. number .. ".ogg"
-			local sleeptime = Vocalization[voice]["z"..number]
-			
-			-- Is the score a Quad Star? If so, we might need to pick one of the
-			-- many available Quad Star soundbytes available for this voice.
-			if number == "quad" then
-				local NumberOfQuads = 0
-				for k,v in pairs(Vocalization[voice]["quad"]) do
-					NumberOfQuads = NumberOfQuads + 1
+				local voice_path= THEME:GetPathO("","Vocalize/" .. voice .. "/default.lua", true)
+				if voice_path ~= "" then
+					dofile(voice_path)
+					voice_loaded= true
+				else
+					voice_loaded= false
 				end
-				local WhichQuad = math.random(NumberOfQuads)
-				sleeptime = Vocalization[voice]["quad"]["z100percent" .. WhichQuad ]
-				number = "100percent" .. WhichQuad
-				soundbyte = GetVocalizeDir() .. voice .. "/" .. number .. ".ogg"
 			end
+			if voice_loaded then
+				local number = Digits[ActiveDigit][1]
+				local soundbyte = GetVocalizeDir() .. voice .. "/" .. number .. ".ogg"
+				local sleeptime = Vocalization[voice]["z"..number]
+
+				-- Is the score a Quad Star? If so, we might need to pick one of the
+				-- many available Quad Star soundbytes available for this voice.
+				if number == "quad" then
+					local NumberOfQuads = 0
+					for k,v in pairs(Vocalization[voice]["quad"]) do
+						NumberOfQuads = NumberOfQuads + 1
+					end
+					local WhichQuad = math.random(NumberOfQuads)
+					sleeptime = Vocalization[voice]["quad"]["z100percent" .. WhichQuad ]
+					number = "100percent" .. WhichQuad
+					soundbyte = GetVocalizeDir() .. voice .. "/" .. number .. ".ogg"
+				end
 
-			SOUND:PlayOnce( soundbyte )
-			self:sleep( sleeptime )
+				SOUND:PlayOnce( soundbyte )
+				self:sleep( sleeptime )
+			end
 		end
 		
 		ActiveDigit = ActiveDigit+1
```

**Comment by dguzek** [comment]
Thanks for bringing this to my attention and for offering a fix via patch.

It looks like there are still edge cases that could sneak through this fix (if a player had *Blender* saved to his/her profile and the theme didn't find any voices, for example), so I'll revisit it another night when I have a bit more time.  :)

**Comment by kyzentun** [comment]
That's a good point, I didn't try it with Blender.

**Comment by dguzek** [comment]
This issue should be resolved via 2f81fbd9a1635fdb9f9bd863ea918024c7acc0ca.
